### DISCUSSION TITLE
Assigning a WM_CLASS atom to a libqtile.bar.Bar object

### DISCUSSION THREAD

**Discussion by Krishna-Sivakumar** [qtile/qtile#2351] [Q&A] [answered]
Hello everyone, this is my first post here!

I'd like to exclude the status bar from a few picom configurations. 
Specifically, I'm trying to exclude the qtile bar from the rounded-corners list. 

I find that using a `WM_CLASS` to identify the window to be excluded is the preferred method for doing so, but I haven't been able to find any methods to do so. 

Any ideas here? Is there some other way to exclude _**specifically**_ the bar from picom? 
Thanks a lot in advance!

**Comment by tych0** [comment, marked as answer]
I don't see any reason we couldn't add a WM_CLASS to the bar's window. I'd be happy to take a patch with a test that adds one.

**Reply by Krishna-Sivakumar** [reply]
Any pointers as to where I could start patching this in?

**Reply by elParaguayo** [reply]
The bar window is created here:
https://github.com/qtile/qtile/blob/3e899ddb493b6bd6c2061a40a62e77949438ea33/libqtile/bar.py#L213-L218

So, after it's created, you just need to add a line to set the property. From what I can tell, you'd add a line like this:
``` python
self.window.window.set_property("WM_CLASS", "qtile_bar", type="STRING", format=8)
```

**Reply by Krishna-Sivakumar** [reply]
Thanks, I'll check if this works for my problem.

**Comment by Barbaross93** [comment]
This is what I use to exclude the bar:
```
  "QTILE_INTERNAL:32c = 1"
```
The bar is considered a cardinal window which is why we need the funny 32c bit

**Reply by elParaguayo** [reply]
One downside with that is that would catch all internal windows which includes popups. May not be ideal if you want to differentiate between them.

**Reply by Barbaross93** [reply]
Technically yes, but I haven't noticed this rule affecting other windows. I could definitely see this applying to those new bar mixins/tooltips you've been cooking up
